quantus/metrics/localisation_metrics.py

```diff
@@ -1220,3 +1221,241 @@ def __call__(
+            kwargs: Keyword arguments (optional)
+                channel_first (boolean): Indicates of the image dimensions are channel first, or channel last.
+                Inferred from the input shape by default.
+                explain_func (callable): Callable generating attributions, default=Callable.
+                device (string): Indicated the device on which a torch.Tensor is or will be allocated: "cpu" or "gpu",
+                default=None.
+
+        Returns
+            last_results: a list of floats with the evaluation outcome of concerned batch
+
+        Examples
+            # Enable GPU.
+            >> device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+
+            # Load a pre-trained LeNet classification model with adaptive pooling layer (architecture at quantus/helpers/models).
+            >> model = LeNetAdaptivePooling(input_shape=(1, 28, 28))
+            >> model.load_state_dict(torch.load("assets/mnist", map_location="cpu", pickle_module=pickle))
+
+            # Load MNIST datasets and make loaders.
+            >> transformer = transforms.Compose([transforms.ToTensor()])
+            >> test_set = torchvision.datasets.MNIST(root='./sample_data', train=False, transform=transformer, download=True)
+            >> test_loader = torch.utils.data.DataLoader(test_set, batch_size=32, pin_memory=True)
+
+            # Load a batch of inputs and outputs to use for XAI evaluation.
+            >> x_batch, y_batch = iter(test_loader).next()
+            >> x_batch, y_batch = x_batch.cpu().numpy(), y_batch.cpu().numpy()
+
+            # Create mosaics (for example 10 mosaics per class)
+            >> x_mosaic_batch, _, _, p_mosaic_batch, y_mosaic_batch = mosaic_creation(images=x_batch,
+                                                                       labels=y_batch,
+                                                                       mosaics_per_class=10,
+                                                                       seed=777)
+
+            # Initialise the metric and evaluate explanations by calling the metric instance.
+            >> metric = Focus()
+            >> scores = {method: metric(model=model,
+                           x_batch=x_mosaic_batch,
+                           y_batch=y_mosaic_batch,
+                           a_batch=None,
+                           p_batch=p_mosaic_batch,
+                           **{"explain_func": explain,
+                              "method": "GradCAM",
+                              "gc_layer": "model._modules.get('conv_2')",
+                              "pos_only": True,
+                              "interpolate": (2*28, 2*28),
+                              "interpolate_mode": "bilinear",
+                              "device": device}) for method in ["GradCAM", "IntegratedGradients"]}
+
+            # Plot example!
+            >> metric.plot(results=scores)
+        """
+
+        if a_batch is None:
```

Comment:
Please put this under line 1388 where we already have a check if a_batch is None.